x/gamm/keeper/grpc_query.go

```diff
@@ -415,3 +420,13 @@ func (q Querier) TotalLiquidity(ctx context.Context, _ *types.QueryTotalLiquidit
 		Liquidity: q.Keeper.GetTotalLiquidity(sdkCtx),
 	}, nil
 }
+
+// Deprecated: use EstimateSwapExactAmountIn in x/swaprouter.
+func (q Querier) EstimateSwapExactAmountIn(ctx context.Context, req *types.QuerySwapExactAmountInRequest) (*types.QuerySwapExactAmountInResponse, error) {
+	panic("not implemented")
+}
+
+// Deprecated: use EstimateSwapExactAmountOut in x/swaprouter.
+func (q Querier) EstimateSwapExactAmountOut(ctx context.Context, req *types.QuerySwapExactAmountOutRequest) (*types.QuerySwapExactAmountOutResponse, error) {
+	panic("not implemented")
+}
```

Comment:
Can you explain this a little further in regards to future steps? Do these panic because they are not client facing?

Reply:
Here I'm only bringing back the client-facing APIs. The routing to swap router will be handled in the next PR